src/test/regress/citus_tests/run_test.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+import pathlib
+from glob import glob
+import argparse
+import shutil
+import random
+import re
+import common
+
+args = argparse.ArgumentParser()
+args.add_argument("-t", "--test", required=True, help="Relative path for test file (must have a .sql or .spec extension)", type=pathlib.Path)
+args.add_argument("-n", "--ntimes", required=True, help="Number of test to run", type=int)
+args.add_argument("-s", "--schedule", required=False, help="Test schedule to be used as a base (optional)", nargs='?', const='', default='')
+
+args = vars(args.parse_args())
+
+regress_dir = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))
+test_file_path = os.path.join(os.getcwd(), args['test'])
+
+if not os.path.isfile(test_file_path):
+    print(f"ERROR: test file '{test_file_path}' does not exist")
+    sys.exit(2)
+
+test_file_extension = pathlib.Path(test_file_path).suffix
+test_file_name = pathlib.Path(test_file_path).stem
+
+if not test_file_extension in '.spec.sql':
+    print(
+        "ERROR: Unrecognized test extension. Valid extensions are: .sql and .spec"
+    )
+    sys.exit(1)
+
+test_schedule = ''
+
+# find related schedule
+for schedule_file_path in sorted(glob(os.path.join(regress_dir, "*_schedule"))):
+        for schedule_line in open(schedule_file_path, 'r'):
+            if  re.search(r'\b' + test_file_name + r'\b', schedule_line):
+                test_schedule = pathlib.Path(schedule_file_path).stem
+                test_schedule_line = schedule_line
+                break
+        else:
+            continue
+        break
```

Comment:
Using the whole test schedule line should require a flag to be passed to `run_test.py`. It's good for flaky test detection, but for development you usually only want to run your current test.

Reply:
I think using the whole line in the schedule is good to reproduce the exact scenario in the development environment, as well. And if you want to run your current test individually, you can use a temp schedule that contains only the test you want to run. 

I think that we can leave this one to the next iteration.